File: src/services/sjFestival.ts
import { createService } from "../common.js";

/**
 * Creates an Express middleware for proxying requests to the Sejong Festival API.
 * Access is restricted to a specific set of endpoints.
 */
export default function createSjFestival() {
  const baseUrl = "https://apis.data.go.kr/5690000/sjFestival";
  const allowedPaths = new Set(["/sj_00000360"]);
  return createService(baseUrl, allowedPaths);
}

File: internal/services/sj_festival.go
package services

var SjFestivalSpec = ServiceSpec{
	MountPath: "/sjFestival",
	BaseURL:   "https://apis.data.go.kr/5690000/sjFestival",
	AllowedPaths: []string{
		"/sj_00000360",
	},
}
